lib/backup_restore/logger.rb

```diff
@@ -27,6 +27,30 @@ def log(message, ex = nil)
       end
     end
 
+    def self.save_log_to_upload(user:, filename: 'log.txt', logs:)
+      Dir.mktmpdir do |dir|
+        logfile = File.new(File.join(dir, filename), 'w')
+        logfile.write(Discourse::Utils.pretty_logs(logs))
+        logfile.close
+
+        zipfile = Compression::Zip.new.compress(dir, filename)
+        upload = File.open(zipfile) do |file|
+          UploadCreator.new(
+            file,
+            File.basename(zipfile),
+            type: 'backup_logs',
+            for_export: 'true'
+          ).create_for(user.id)
+        end
+
+        if !upload.persisted?
+          Rails.logger.warn("Failed to upload the backup logs file #{zipfile}")
```

Comment:
I think we should only warn in our logs if the warning can help developers debug the problem. Otherwise, adding to the logs is just going to add noise to the logs.

Reply:
I added upload error messages.